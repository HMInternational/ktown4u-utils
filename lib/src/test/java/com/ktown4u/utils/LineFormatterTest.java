package com.ktown4u.utils;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class LineFormatterTest {
    @Test
    void line_with_no_depth() {
        LineFormatter lineFormatter = new LineFormatter(80);
        String formattedLine = lineFormatter.formatLineWithWhitespaces("name", "value");
        String expected = "name                                                                       value\n";
        assertThat(formattedLine).isEqualTo(expected);
    }

    @Test
    void line_with_depth() {
        LineFormatter lineFormatter = new LineFormatter(80);
        String formattedLine = lineFormatter.formatLineWithWhitespaces(1, "name", "value");
        String expected = "    name                                                                   value\n";
        assertThat(formattedLine).isEqualTo(expected);
    }

    @Test
    void multiple_lines_with_depth() {
        LineFormatter lineFormatter = new LineFormatter(80);
        List<IndentiedLine> lines = List.of(
                new IndentiedLine(1, "key1", "value1"),
                new IndentiedLine(1, "key2", "value2"),
                new IndentiedLine(2, "key11", "value1"),
                new IndentiedLine(2, "key12", "value2")
        );
        String formattedLine = lineFormatter.formatLineWithWhitespaces("key", null);
        for (IndentiedLine line : lines) {
            formattedLine += lineFormatter.formatLineWithWhitespaces(line.depth, line.name, line.value);
        }
        Approvals.verify(formattedLine);
    }

    record IndentiedLine(int depth, String name, String value) {
    }
}