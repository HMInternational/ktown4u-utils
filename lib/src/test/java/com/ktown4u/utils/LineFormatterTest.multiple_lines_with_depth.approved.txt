key                                                                         null
    key1                                                                  value1
    key2                                                                  value2
        key11                                                             value1
        key12                                                             value2
